source/addonHandler.py

```diff
@@ -35,6 +37,11 @@
 NVDA_ADDON_PROG_ID = "NVDA.Addon.1"
 ADDON_PENDINGINSTALL_SUFFIX=".pendingInstall"
 DELETEDIR_SUFFIX=".delete"
+#: Compiled regular expression to match an NVDA version string.
+#: Supports year.month versions (e.g. 2018.1) and an additional suffix for bug fix releases (e.g. 2018.1.1)
+#: It also matches against strings like 2018.1dev
+#: @type: RegexObject
+NVDA_VERSION_REGEX = re.compile(r"(\d{4,4})\.(\d)(?:\.(\d))?")
```

Comment:
Some automated tests for the regex would be handy. Makes it a lot easier for someone who may have to modify it in the future to be sure that they haven't regressed for any of the cases that have already been considered.